flink-runtime/src/main/java/org/apache/flink/runtime/state/delegate/DelegateKeyedStateBackend.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.state.delegate;
+
+import org.apache.flink.api.common.state.CheckpointListener;
+import org.apache.flink.runtime.state.AbstractKeyedStateBackend;
+import org.apache.flink.runtime.state.CheckpointableKeyedStateBackend;
+
+/**
+ * {@link DelegateKeyedStateBackend} wraps an underlying keyed state backend, and delegates its
+ * partitioned state access.
+ *
+ * @param <K> Type of the key by which states are keyed.
+ */
+public interface DelegateKeyedStateBackend<K>
+        extends CheckpointableKeyedStateBackend<K>, CheckpointListener {
+
+    AbstractKeyedStateBackend<K> getDelegatedKeyedStateBackend();
```

Comment:
1. As `StateConfigUtil.isStateImmutableInStateBackend` doesn't need to explicitly access delegated backend, this interface becomes unnecessary.
2. If not, I think it should be `@Experimental`
3. And `getDelegatedKeyedStateBackend` should return `KeyedStateBackend`

Reply:
1. As the reason mentioned above, it still needs to access delegated backend
2. Why need to add `@Experimental`? Do we want to remove it later? Do we need to add `@Experimental` in other places?
3. And getDelegatedKeyedStateBackend should return KeyedStateBackend (OK)

Reply:
So the interface is removed, right?

Reply:
Right.